woocommerce/woocommerce
# Setup the product editor to support the product variation

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

The base of this PR is adding support for using the product editor with multiple product types, as we previously had hardcoded the `product` post type in a lot of places. This would not work for loading a variation specific form.
In addition, this PR also adds a new page for loading a variation form and adds a new template for product variations.

**Note:**
This is the first of several PRs to make editing variations possible within the new editor, this PR updates most of the main items to get a product variation template loaded.

Closes https://github.com/woocommerce/woocommerce/issues/40573



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a WooCommerce store with this branch and enable the product editor using **WooCommerce > Settings > Advanced > Features**
2. Go to **Products > Add New**
3. Add a name and click save draft, now go to the variations tab and add several variation options.
4. Wait until the variations are generated.
5. Click on one of the 3 dot menus of a variation a take note of the **Variation Id** listed at the top.
6. Now update your URL removing the `&tab=variations` at the end and replacing it with `/variation/<variation_id>` (replacing the variation id with the one from the 3 dot menu )
7. This should load a new template with a General, Pricing, Inventory, and Shipping tab.
8. The General only contains a Variation Details and Images section. The variation details contains a note field.
9. Type something in the note field and click update
10. Updating should work correctly and the **Note** value should persist on refresh (Note: **Note** is the only field that will work at this moment ).



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
